adala/utils/llm.py

```diff
@@ -41,6 +42,34 @@ class ErrorLLMResponse(LLMResponse):
     adala_details: str = Field(default=None, serialization_alias='_adala_details')
 
 
+class LiteLLMInferenceSettings(BaseSettings):
+    '''
+    Common inference settings for LiteLLM.
+
+    Attributes:
+        model: model name. Refer to litellm supported models for how to pass
+               this: https://litellm.vercel.app/docs/providers
+        api_key: API key, optional. If provided, will be used to authenticate
+                 with the provider of your specified model.
+        base_url (Optional[str]): Base URL, optional. If provided, will be used to talk to an OpenAI-compatible API provider besides OpenAI.
+        api_version (Optional[str]): API version, optional except for Azure.
+        instruction_first: Whether to put instructions first.
+        response_model: Pydantic model to constrain the LLM generated response. If not provided, the raw completion text will be returned.  # noqa
+        max_tokens: Maximum tokens to generate.
+        temperature: Temperature for sampling.
+        timeout: Timeout in seconds.
+        seed: Integer seed to reduce nondeterminism in generation.
+    '''
+    model: str = 'gpt-4o-mini'
+    api_key: Optional[str] = None
+    base_url: Optional[str] = None
+    api_version: Optional[str] = None
+    max_tokens: int = 1000
+    temperature: float = 0.0
+    timeout: Optional[Union[float, int]] = None
+    seed: Optional[int] = 47
```

Comment:
not 42 or 1337? 🤯

Reply:
https://memory-alpha.fandom.com/wiki/47 :face_with_peeking_eye:

Reply:
ah, acceptable lol